superset/db_engine_specs.py

```diff
@@ -114,23 +114,31 @@ def get_limit_from_sql(cls, sql):
                 (?ix)          # case insensitive, verbose
                 \s+            # whitespace
                 LIMIT\s+(\d+)  # LIMIT $ROWS
-                ;?             # optional semi-colon
-                (\s|;)*$       # remove trailing spaces tabs or semicolons
+                .*$            # everything else
                 """)
         matches = limit_pattern.findall(sql)
         if matches:
-            return int(matches[0][0])
+            return matches[0]
 
     @classmethod
     def get_query_without_limit(cls, sql):
-        return re.sub(r"""
+        before_limit = re.sub(r"""
```

Comment:
You probably should rename the function given the change in return values. Also can we make these staticmethods.